hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapClearMessageTask.java

```diff
@@ -46,16 +49,22 @@ protected OperationFactory createOperationFactory() {
 
     @Override
     protected Object reduce(Map<Integer, Object> map) {
-        int totalAffectedEntries = 0;
+        int clearedTotal = 0;
         for (Object affectedEntries : map.values()) {
-            totalAffectedEntries += (Integer) affectedEntries;
+            clearedTotal += (Integer) affectedEntries;
         }
-        final MapService service = getService(MapService.SERVICE_NAME);
-        final Address thisAddress = service.getMapServiceContext().getNodeEngine().getThisAddress();
-        if (totalAffectedEntries > 0) {
-            service.getMapServiceContext().getMapEventPublisher()
-                    .publishMapEvent(thisAddress, parameters.name, EntryEventType.CLEAR_ALL, totalAffectedEntries);
+
+        MapService service = getService(MapService.SERVICE_NAME);
+        MapServiceContext mapServiceContext = service.getMapServiceContext();
+
+        if (clearedTotal > 0) {
```

Comment:
Should we send the event regardless of this number? It is possible that the cache became empty after client called clear before the response is returned to the client. Therefore, it will be safer to send this event regardless.